app/PrintAccount.php

```diff
@@ -26,4 +31,22 @@ class PrintAccount extends Model
     public function user() {
         return $this->belongsTo('App\User');
     }
+
+    public function hasEnoughMoney($balance) {
+        return $this->balance >= abs($balance);
+    }
+
+    public static function getCost($pages, $twosided, $number_of_copies = 1) {
+        if ($twosided) {
+            $cost;
+            if ($pages % 2 == 0) {
+                $cost = $pages / 2 * PrintAccount::COST['twosided'];
+            } else {
+                $cost = ($pages  - 1) / 2 * PrintAccount::COST['twosided'] + $pages * PrintAccount::COST['onesided'];
+            }
+            return $cost * $number_of_copies;
+        } else {
+            return $pages * PrintAccount::COST['onesided'] * $number_of_copies;
+        }
+    }
```

Comment:
I'm not convinced that it's correct. Why would you add the `$pages * PrintAccount::COST['onesided']`? There is not at most **one** orphan ending page?
Also the controlflow is pretty ugly.

I would suggest to implement this like so:
```php
public static function getCost($pages, $twosided, $number_of_copies = 1) {
    if (!$twosided)
        return $pages * PrintAccount::COST['onesided'] * $number_of_copies;

    $has_orphan_ending = $pages % 2 == 1;
    $one_copy_cost = floor($pages / 2) * PrintAccount::COST['twosided']
        + $has_orphan_ending * PrintAccount::COST['onesided'];
    return $one_copy_cost * $number_of_copies;
}
```

Reply:
@steakhal +1

Reply:
I'm not sure what does this mean, but probably good :D
Btw, is this proposed logic correct? @vendi95

Reply:
I tried to express my full agreement in a short form. :)
Yes, your logic seems fine to me, not like the one in the PR.

Reply:
Agree, this looks terrible.